scipy/optimize/optimize.py

```diff
@@ -2665,6 +2676,11 @@ def _minimize_powell(func, x0, args=(), callback=None,
         msg = _status_message['maxiter']
         if disp:
             print("Warning: " + msg)
+    elif np.isnan(fval) or np.isnan(x).any():
```

Comment:
likewise this line never evaluates to `True` with current tests

Reply:
Added bailout condition to the main loop, if nan result was obtained both in two consecutive iterations.